### PR TITLE
Fixing Advanced logger wrapping unit test

### PR DESCRIPTION
**Pull request by kuqin12**
# Preface

Please ensure you have read the [contribution docs](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md) prior
to submitting the pull request. In particular,
[pull request guidelines](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md#pull-request-best-practices).

## Description

The existing shell test will not take the disabled processors into account, thus causing the system to have false positive results.

This change updated the logic to tick out disabled cores, removed some UT logs to avoid spamming the output and replaced all the returned status with UT_ASSERT for easier troubleshooting.

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [ ] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [x] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

This change was tested on proprietary hardware with multiple cores.

## Integration Instructions

N/A

### CODE REVIEW

File: AdvLoggerPkg/UnitTests/AdvancedLoggerWrapper/AdvancedLoggerWrapperTestApp.c [hunk 1]
```diff
@@ -185,6 +185,7 @@ InitializeInMemoryLog (
 
     if (!ValidateInfoBlock ()) {
       mLoggerInfo = NULL;
+      UT_ASSERT_TRUE (FALSE);
```
Comment (os-d) [inline AdvLoggerPkg/UnitTests/AdvancedLoggerWrapper/AdvancedLoggerWrapperTestApp.c]: nit: 
```suggestion
      UT_ASSERT_NOT_NULL (mLoggerInfo);

```

Just for when reading the log

File: AdvLoggerPkg/UnitTests/AdvancedLoggerWrapper/AdvancedLoggerWrapperTestApp.c [hunk 2]
```diff
@@ -295,28 +298,27 @@ TestCursorWrappingMP (
...
+  UINTN                      NumberOfProcessors;
+  UINTN                      EnabledProcessors;
+  UINT8                      *TempCache = NULL;
+  UINTN                      PrefixSize;
+  CHAR8                      EndChar;
+  EFI_PROCESSOR_INFORMATION  CpuInfo;
 
   Btc = (BASIC_TEST_CONTEXT *)Context;
 
+  UT_ASSERT_TRUE (mLoggerInfo != NULL);
```
Comment (os-d) [inline AdvLoggerPkg/UnitTests/AdvancedLoggerWrapper/AdvancedLoggerWrapperTestApp.c]: small nit, for readability:
```suggestion
  UT_ASSERT_NOT_NULL (mLoggerInfo);

```

File: AdvLoggerPkg/UnitTests/AdvancedLoggerWrapper/AdvancedLoggerWrapperTestApp.c [hunk 3]
```diff
@@ -392,47 +389,31 @@ TestCursorWrappingMP (
...
-
-    if (TempCache[StrIndex] == 0) {
-      // Printed this index already
-      UtStatus = UNIT_TEST_ERROR_TEST_FAILED;
-      goto Done;
-    } else {
-      TempCache[StrIndex] = 0;
-    }
+    EndChar                                                           = mMessageEntry.Message[ADV_TIME_STAMP_PREFIX_LEN + PrefixSize + 8];
+    mMessageEntry.Message[ADV_TIME_STAMP_PREFIX_LEN + PrefixSize + 8] = '\0';
```
Comment (os-d) [inline AdvLoggerPkg/UnitTests/AdvancedLoggerWrapper/AdvancedLoggerWrapperTestApp.c:396]: nit: if you move this line above the previous line and add an empty line on either side, we won't have the way too far to the right formatting for these other lines.

**Review by os-d — APPROVED** [review]

File: AdvLoggerPkg/UnitTests/AdvancedLoggerWrapper/AdvancedLoggerWrapperTestApp.c (hunk 3 above)
Comment (kuqin12) [reply AdvLoggerPkg/UnitTests/AdvancedLoggerWrapper/AdvancedLoggerWrapperTestApp.c:396]: Move this line up will change the logic order. But I will add an empty line between this line and the one above to see how that goes. Will update here shortly.

Comment (kuqin12) [reply AdvLoggerPkg/UnitTests/AdvancedLoggerWrapper/AdvancedLoggerWrapperTestApp.c:396]: Updated by adding some comments between the lines :)

Comment (os-d) [reply AdvLoggerPkg/UnitTests/AdvancedLoggerWrapper/AdvancedLoggerWrapperTestApp.c:396]: Oh, missed that, I just looked and saw it wasn't using EndChar.

**Review by kenlautner — APPROVED** [review]

**Pull request merged**